pkg/divert/nginx/services.go

```diff
@@ -81,20 +62,16 @@ func translateService(name, namespace string, s *apiv1.Service) *apiv1.Service {
 			Labels:      s.Labels,
 			Annotations: s.Annotations,
 		},
-		Spec: s.Spec,
+		Spec: apiv1.ServiceSpec{
+			Type:         apiv1.ServiceTypeExternalName,
+			ExternalName: fmt.Sprintf("%s.%s.svc.cluster.local", s.Name, s.Namespace),
+		},
 	}
 	labels.SetInMetadata(&result.ObjectMeta, model.DeployedByLabel, format.ResourceK8sMetaString(name))
-	// create a headless service pointing to an endpoints object that resolves to service cluster ip in the diverted namespace
-	result.Spec.ClusterIP = apiv1.ClusterIPNone
-	result.Spec.ClusterIPs = nil
-	result.Spec.Selector = nil
 	if result.Annotations == nil {
 		result.Annotations = map[string]string{}
 	}
 	result.Annotations[model.OktetoAutoCreateAnnotation] = "true"
+	result.Annotations[model.OktetoDivertedSourceAnnotation] = s.Namespace
```

Comment:
Just curiosity, why is it needed in the service? I see that in the backend we are only checking the annotation for the ingress.

Reply:
👍  Ok, thanks for the clarification. No problem on adding it, just curiosity